Updated PreprCMS guide (#6983)
withastro/docs: src/content/docs/en/guides/cms/preprcms.mdx

```astro title="src/pages/index.astro" ins={15-23}

---
import Layout from '../layouts/Layout.astro';
import {Prepr} from '../lib/prepr.js';
import GetArticles from '../queries/get-articles.js';

const response = await Prepr(GetArticles)
const { data } = await response.json()
const articles = data.Articles
---

<Layout title="My blog site">
    <h1>
        My blog site
    </h1>
    <ul>
        {
            articles.items.map((post) => (
                <li>
                    {post.title}
                </li>
            ))
        }
    </ul>
</Layout>
```

Your root directory should include these new files:

<FileTree title="Project Structure">
- lib/
    - prepr.js
    - **queries**/
        - **get-articles.js**
- src/
    - pages/
      - index.astro
- .env
- astro.config.mjs
- package.json
</FileTree>

When you view the website on your localhost, you'll see something like the image below.

![blog article list](https://assets-site.prepr.io//770picl832e3-screenshot-2023-05-10-at-105008.png)

Now that we have a list of articles, let's expand this example into a full blog where every item's link lets you view the entire blog article.

### Retrieving content item details

> So far, you've fetched multiple content items, but to create a fully working blog you also need to be able to fetch single content items. Once you have the list of articles in your overview page, add links to them. This enables a visitor to open a detailed article page.
> Update the `index.astro` file to include a link on each article title as shown in the code below.
> 
> ```astro title="src/pages/index.astro" ins="<a href={post._slug}>{post.title}</a>" 
> 
> ---
> import Layout from '../layouts/Layout.astro';
> import {Prepr} from '../lib/prepr.js';
> import GetArticles from '../queries/get-articles.js';
> 
> const response = await Prepr(GetArticles)
> const { data } = await response.json()
> const articles = data.Articles
> ---
> 
> <Layout title="Welcome to Astro.">
>     <h1>
>         My blog site
>     </h1>
>     <ul>
>         {
>             articles.items.map((post) => (
>                 <li>
>                     <a href={post._slug}>{post.title}</a>
>                 </li>
>             ))
>         }
>     </ul>
> </Layout>
> ```
> 
> Now when you view the page, each article has its own link. Continue with the next step to fetch the article details.

Reviewer: ```suggestion
```
Let's combine this step with the one above. There's not much new going on here!

> Add another query to fetch a specific article by its slug and make this page visible when a visitor clicks an article link.
> 1. Create a file called `get-article-by-slug.js` in the `queries` folder and add the following query to this file to query a specific article by its slug:

Reviewer: ```suggestion
To create a page for each blog post, you will execute a new GraphQL query on a [dynamic routing]() `.astro` page. This query will fetch a specific article by its slug and a new page will be created for each individual blog post.

1. Create a file called `get-article-by-slug.js` in the `queries` folder and add the following to query a specific article by its slug and return data such as the article `title` and `content`:
```

```js title="src/lib/queries/get-article-by-slug.js"

const GetArticleBySlug = `
query ($slug: String) {
   Article (slug: $slug) {
     _id
     title
     content {
       __typename
       ... on Text {
         body
         text
       }
       ... on Assets {
         items {
           url
         }
       }
     }
   }
}`

export default GetArticleBySlug
```

> Now that the query is added, fetch the individual article by its slug. Fetch the article title and the article content.
> 

Reviewer: ```suggestion

```

:::tip
You can create and [test GraphQL queries](https://docs.prepr.io/reference/graphql/v1/test-queries) using the [Apollo explorer](https://studio.apollographql.com/sandbox/explorer) in Prepr. Open the API Explorer from the *Article* content item page in Prepr.
The Article content is stored in a *Dynamic content field*. Check out the GraphQL docs for more details on [how to fetch the data within this field](https://docs.prepr.io/reference/graphql/v1/schema-field-types-dynamic-content-field).
:::

> 
> 2. Go to the `src/pages` folder and create a file called `[…slug].astro` file with the following code to execute the query that you just created and display the retrieved article:

Reviewer: ```suggestion
2. Inside the `src/pages` folder, create a file called `[…slug].astro`. Add the following code to import and execute the query from the previous step and display the retrieved article:
```

```astro title="src/pages/[...slug].astro"

> 

Reviewer: ```suggestion
```

---
import Layout from '../layouts/Layout.astro';
import {Prepr} from '../lib/prepr.js';
import GetArticleBySlug from '../queries/get-article-by-slug.js';

const { slug } = Astro.params;
const response = await Prepr(GetArticleBySlug, {slug})
const { data } = await response.json()
const article = data.Article
---

> <Layout title={article.title}>
>     <main>
>         <h1>{article.title}</h1>
>         {
>             article.content.map((content) => (
>                 <div>
>                     {
>                         content.__typename === "Assets" &&
>                         <img src={content.items[0].url} width="300" height="250"/>
>                     }
>                     {
>                         content.__typename === 'Text' &&
>                         <div set:html={content.body}></div>
>                     }
>                 </div>
>             ))
>         }
>     </main>
> </Layout>

Reviewer: ```suggestion
<Layout title={article.title}>
  <main>
    <h1>{article.title}</h1>
    {
      article.content.map((content) => (
        <div>
          {
            content.__typename === "Assets" &&
            <img src={content.items[0].url} width="300" height="250"/>
          }
          {
            content.__typename === 'Text' &&
            <div set:html={content.body}></div>
          }
        </div>
      ))
    }
  </main>
</Layout>
```

```

> Your root directory should include these new files:
> 
> <FileTree title="Project Structure">
> - lib/
>     - prepr.js
>     - queries/
>         - get-articles.js
>         - **get-article-by-slug.js**
> - src/
>     - pages/
>       - index.astro
>       - **[…slug].astro**
> - .env
> - astro.config.mjs
> - package.json
> </FileTree>

Reviewer: ```suggestion
Your root directory should now include these new files:

<FileTree title="Project Structure">
- lib/
    - prepr.js
    - queries/
        - get-articles.js
        - **get-article-by-slug.js**
- src/
    - pages/
      - index.astro
      - **[…slug].astro**
- .env
- astro.config.mjs
- package.json
</FileTree>
```
Love that the file tree is added directly after the new file is created here!

> Now, when you view your site, click an article to open the article detail page like in the image below.

Reviewer: ```suggestion
Now, when you click an article link from the main list of blog posts, you will be taken to a page with its individual content.
```

> ![blog article detail](https://assets-site.prepr.io//507hx87v7wkw-screenshot-2023-05-11-at-162824.png)
> 
> Congratulations! You've successfully integrated an Astro project with Prepr for a simple blog.

Reviewer: ```suggestion
### Publishing your site

To deploy your Prepr blog, visit our [deployment guides](/en/guides/deploy/) and follow the instructions for your preferred hosting provider.
```
